src/test/java/lambda/part3/exercise/Exercise4.java

```diff
@@ -0,0 +1,125 @@
+package lambda.part3.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@SuppressWarnings({"unused", "ConstantConditions"})
+class Exercise4 {
+
+    private static class LazyCollectionHelper<T, R> {
+
+        private final List<T> list;
+        private final Function<List<T>, List<R>> mapet;
+
+       private  LazyCollectionHelper(List<T> list, Function<List<T>, List<R>> mapet){
+            this.list = list;
+            this.mapet = mapet;
+        }
+
+        public static <T> LazyCollectionHelper<T, T> from(List<T> list) {
+            return new LazyCollectionHelper<> (list, x->x);
+        }
+
+        public <U> LazyCollectionHelper<T, U> flatMap(Function<R, List<U>> flatMapping) {
+            Function<List<R>, List<U>> nextMapet = listR-> {
+                List<U> listU = new ArrayList<>();
+                listR.forEach(r->listU.addAll(flatMapping.apply(r)));
+                return listU;
+            };
+            return new LazyCollectionHelper<>(list, mapet.andThen(nextMapet));
+        }
+
+        public <U> LazyCollectionHelper<T, U> map(Function<R, U> mapping)
+        {
+            Function<List<R>, List<U>> nextMapet = listR-> {
+                List<U> listU = new ArrayList<>();
+                listR.forEach(r -> listU.add(mapping.apply(r)));
+                return listU;
```

Comment:
This string is only difference between map and flatMap methods, it's better to extract code duplicates in another method